pkg/ccl/changefeedccl/kafkaauth/kafkaauth.go

```diff
@@ -0,0 +1,163 @@
+package kafkaauth
+
+import (
+	"context"
+
+	"github.com/IBM/sarama"
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeedbase"
+	"github.com/cockroachdb/errors"
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+type saslMechanismBuilder interface {
+	name() string
+	validateParams(u *changefeedbase.SinkURL) error
+	build(u *changefeedbase.SinkURL) (saslMechanism, error)
+}
+
+type saslMechanism interface {
```

Comment:
Given this interface type is returned by exported functions, maybe it should also be exported and documented?